src/codec.cairo

```diff
@@ -29,13 +31,70 @@ pub impl EncodeByteArrayImpl of Encode<@ByteArray> {
         dest.append(self);
     }
 }
+pub impl EncodeTxIn of Encode<TxIn> {
+    fn encode_to(self: TxIn, ref dest: ByteArray) {
+        encode_compact_size(self.script.len(), ref dest);
+        dest.append(self.script);
+        encode_compact_size(self.sequence, ref dest);
+        self.previous_output.encode_to(ref dest);
+        encode_compact_size(self.witness.len(), ref dest);
+        // now we have to serialize the witness
+    }
+}
+
+pub impl EncodeTxOut of Encode<TxOut> {
+    fn encode_to(self: TxOut, ref dest: ByteArray) {
+        encode_compact_size(self.value.try_into().unwrap(), ref dest);
+        encode_compact_size(self.pk_script.len(), ref dest);
+        dest.append(self.pk_script);
+    }
+}
+
+pub impl EncodeOutpoint of Encode<OutPoint> {
+    fn encode_to(self: OutPoint, ref dest: ByteArray) {
+        let txids: ByteArray = self.txid.into();
+        let txids_rev = txids.rev();
+        dest.append(@txids_rev);
+        encode_compact_size(self.vout, ref dest);
+        encode_compact_size(self.block_time, ref dest);
+    }
+}
+
+
+pub impl EncodeTx of Encode<Transaction> {
+    fn encode_to(self: Transaction, ref dest: ByteArray) {
+        encode_compact_size(self.version, ref dest);
+        // if we have to find the wxtid then we have to serilaize according to the segeit
+        // transaction
+        encode_compact_size(self.inputs.len(), ref dest);
+        let inputs: Span<TxIn> = self.inputs;
```

Comment:
Encode is already implemented for Span<T: Encode>

Reply:
using  ```self.inputs.ecode_to(ref dest);``` pump the error -- `ecode_to not found on type core::array::Span::<raito::state::TxIn>`

Reply:
Try change the EncodeSnapImpl to:

```
pub impl EncodeSpanImpl<T, +Encode<T>, +Copy<T>> of Encode<Span<T>> {
    fn encode_to(self: Span<T>, ref dest: ByteArray) {
        encode_compact_size(self.len(), ref dest);
        for item in self {
            (*item).encode_to(ref dest);
        }
    }
}
```

Reply:
Or better implement `Encode<TxIn>` (same goes for other structures)